pyproject.toml

```diff
@@ -13,7 +13,7 @@ dependencies = [
     "importlib-metadata",
     "packaging",
     "pandas>=1.2.4",
-    "pyarrow>=3.0.0",
+    "pyarrow==15.0.0",
```

Comment:
Thanks for updating this! 

Though I think we may need to pin to `16.0.0` instead?
* I see that it is the [pyarrow 16.0 release](https://arrow.apache.org/release/16.0.0.html) which has `[GH-39532](https://github.com/apache/arrow/issues/39532) - [Python] Compatibility with NumPy 2.0` in it
* And on the [[Python] Compatibility with NumPy 2.0](https://github.com/numpy/numpy/issues/26191) issue it lists `16.0` as well:

<img width="827" alt="Screenshot 2024-07-11 at 2 25 38 PM" src="https://github.com/TileDB-Inc/TileDB-Cloud-Py/assets/1396242/b06132f0-9b40-4718-bc94-a6f3d2fc7cd7">

(Unsure) Also, should it be `>=` rather than `==`?

Reply:
Didn't knew that, as both versions are supposedly working with numpy v1.16.1 and above... Will update

Reply:
Does that mean that users of the package will need to have `pyarrow==16.0.0` exactly? What would happen if they depend on another package which, for example, requires ``pyarrow>=16.1.0` (which seems reasonable for another package to require if they depend on a bug fix in a non-`16.0.0` release)? 

Or am I misunderstanding this and it's a build-time only requirement? And at runtime they can have whatever version they want?

Reply:
That's a valid point and I am not sure about that.
I will read about it and update you here!

Reply:
@jparismorgan You were right about that, people using tiledb-cloud will need a minimum version of 16.0.0 but it will not block versions higher than that.
Good catch!